lib/models/undertaking.dart

```diff
@@ -0,0 +1,230 @@
+import 'package:flutter/foundation.dart';
+import 'package:planner/common/recurrence.dart';
+import 'package:planner/common/time_management.dart';
+
+class Undertaking {
```

Comment:
what's the point of Undertaking?

Reply:
It's a superclass for the Task and Event class since they share a lot in common and prof want us to remove duplicated code.